lib/libzfs/libzfs_pool.c

```diff
@@ -4189,11 +4234,12 @@ zpool_get_errlog(zpool_handle_t *zhp, nvlist_t **nverrlistp)
 			nvlist_free(nv);
 			goto nomem;
 		}
-		if (nvlist_add_nvlist(*nverrlistp, "ejk", nv) != 0) {
-			nvlist_free(nv);
-			goto nomem;
-		}
-		nvlist_free(nv);
+
+		block_ids = malloc(count * sizeof (uint64_t));
```

Comment:
We need to free these at some point; once they've been added to the nvlist, they can be safely freed, since adding an array to the nvlist does a deep copy.